receiver/receivercreator/observerhandler.go

```diff
@@ -142,30 +145,43 @@ func (obs *observerHandler) OnAdd(added []observer.Endpoint) {
 				resAttrs,
 				env,
 				e,
-				obs.nextConsumer,
+				obs.nextMetricsConsumer,
+				obs.nextLogsConsumer,
 			)
 
 			if err != nil {
 				obs.params.TelemetrySettings.Logger.Error("failed creating resource enhancer", zap.String("receiver", template.id.String()), zap.Error(err))
 				continue
 			}
 
-			rcvr, err := obs.runner.start(
-				receiverConfig{
-					id:         template.id,
-					config:     resolvedConfig,
-					endpointID: e.ID,
-				},
-				discoveredConfig,
-				resourceEnhancer,
-			)
-
+			var rcvr component.Component
+			if resourceEnhancer.nextMetricsConsumer != nil {
```

Comment:
similar issue w/ the factory where we don't want to telemetry types to be mutually exclusive.